src/storage-client/src/controller.rs

```diff
@@ -2501,10 +2514,13 @@ where
     /// - If `self.state.collections` does not have an entry for `global_id`.
     /// - If `IntrospectionType::ShardMapping`'s `GlobalId` is not registered as
     ///   a managed collection.
-    async fn register_shard_mappings<I>(&self, global_ids: I)
+    /// - If diff is any value other than `1` or `-1`.
+    async fn append_shard_mappings<I>(&self, global_ids: I, diff: i64)
```

Comment:
joining the church of differential :sweat_smile:

Reply:
:D